core/domain/skill_services.py

```diff
@@ -246,6 +246,25 @@ def get_skill_summary_by_id(skill_id, strict=True):
         return None
 
 
+def get_skill_descriptions_by_ids(skill_ids):
+    """Returns a list of skill descriptions corresponding to given skill ids.
+
+    Args:
+        skill_ids: str. The list of skill ids.
+
+    Returns:
+        dict. The skill descriptions of skills keyed by their corresponding ids.
+    """
+    skill_summary_models = skill_models.SkillSummaryModel.get_multi(skill_ids)
+    skill_id_to_description_dict = {}
+
+    for skill_summary_model in skill_summary_models:
+        if skill_summary_model is not None:
```

Comment:
So what do we do if it's None -- should some sort of error occur somewhere? Who should be in charge of fixing the error, how would they know about it, and what should they do?

Reply:
Depending on how we would manage the skill to topic link, I think when a skill is deleted, it could be removed from all the topics it is a part of. Currently, how about we update the topic itself when an error occurs here,  to remove that skill from the topic?

Reply:
No, you can't do state-changing operations in a GET. At most you can throw an error. So the question is whether we should handle it silently (and email the admins) or whether we should break execution altogether. Thoughts?

If you handle it silently then you need to deal with this case in the frontend (i.e. the case where skill descriptions for particular skills are missing).

Reply:
I don't think execution needs to be broken altogether, since the topic creator didn't do anything for  this error to happen. How about if we just don't show those skills in the editor? or maybe have description as Skill Deleted, with a button in the editor to remove all such skills?

Reply:
I think having a "Skill Deleted" description seems fine, but maybe add logic that sends an email to the admin to say "fix this". Editors shouldn't have to deal with system errors.

Reply:
Done.